Date: Sat May 17 02:48:24 2014

From: b18c99a27bd460dd80c3e4bbf40da2239e88e6e2 	Sat May 17 02:48:24 2014

Subject: [PATCH] Added gst plugins for video scaling

	Added gst plugins for video scaling

Source: COMCAST

License: LGPL-2.1-or-later

Upstream-Status: Pending

Signed-off-by: rra <ridish.ra@lnttechervices.com>
---


diff -urN gst-plugins-bad-0.10.23.orig/sys/fbdev/gstfbdevsink.c gst-plugins-bad-0.10.23/sys/fbdev/gstfbdevsink.c
--- gst-plugins-bad-0.10.23.orig/sys/fbdev/gstfbdevsink.c	2014-05-16 23:39:16.251688002 +0000
+++ gst-plugins-bad-0.10.23/sys/fbdev/gstfbdevsink.c	2014-05-16 23:40:12.775688210 +0000
@@ -46,7 +46,7 @@
   ARG_XPOS,
   ARG_YPOS
 };
-
+static int IsPosSet=0;
 static void gst_fbdevsink_base_init (gpointer g_class);
 static void gst_fbdevsink_class_init (GstFBDEVSinkClass * klass);
 static void gst_fbdevsink_get_times (GstBaseSink * basesink,
@@ -215,7 +215,8 @@
 
   /* calculate centering and scanlengths for the video */
   fbdevsink->bytespp = fbdevsink->fixinfo.line_length / fbdevsink->varinfo.xres;
-
+ if(!IsPosSet)
+{
   fbdevsink->cx = ((int) fbdevsink->varinfo.xres - fbdevsink->width) / 2;
   if (fbdevsink->cx < 0)
     fbdevsink->cx = 0;
@@ -223,7 +224,7 @@
   fbdevsink->cy = ((int) fbdevsink->varinfo.yres - fbdevsink->height) / 2;
   if (fbdevsink->cy < 0)
     fbdevsink->cy = 0;
-
+}
   fbdevsink->linelen = fbdevsink->width * fbdevsink->bytespp;
   if (fbdevsink->linelen > fbdevsink->fixinfo.line_length)
     fbdevsink->linelen = fbdevsink->fixinfo.line_length;
@@ -255,7 +256,7 @@
         + fbdevsink->cx * fbdevsink->bytespp,
         GST_BUFFER_DATA (buf) + i * fbdevsink->width * fbdevsink->bytespp,
         fbdevsink->linelen);
-
+   //printf("=======fbdevsink_render fbdevsink->cx  %d==fbdevsink->cy   %d===\n",fbdevsink->cx,fbdevsink->cy);
   return GST_FLOW_OK;
 }
 
@@ -328,15 +329,20 @@
       tmp = g_value_dup_string (value);
 	 pos =atoi(tmp);
       if(pos < fbdevsink->varinfo.xres)
+	{
+	  IsPosSet=1;
 	  fbdevsink->cx= pos;
-
+	}
 	  free(tmp);
 	break;
 	case ARG_YPOS:
             tmp = g_value_dup_string (value);
 	   pos =atoi(tmp);
 	  if(pos < fbdevsink->varinfo.yres)
+         {	
+	   IsPosSet=1;
 	  fbdevsink->cy= pos;
+         }
 	  free(tmp);
 	break;
     default:
@@ -428,6 +434,7 @@
   gstvs_class->render = GST_DEBUG_FUNCPTR (gst_fbdevsink_render);
   gstvs_class->start = GST_DEBUG_FUNCPTR (gst_fbdevsink_start);
   gstvs_class->stop = GST_DEBUG_FUNCPTR (gst_fbdevsink_stop);
+ IsPosSet =0;
 
 }
 
